aws/signer/v4/v4.go

```diff
@@ -696,6 +698,81 @@ func (ctx *signingCtx) removePresign() {
 	ctx.Query.Del("X-Amz-SignedHeaders")
 }
 
+// Set ctx.Request.Host to host:port for non-standard ports, and to host for explicit standart ports
+func (ctx *signingCtx) sanitizeHost() {
+	if ctx.Request == nil || ctx.Request.URL == nil {
+		return
+	}
+
+	var port string
+	if ctx.Request.Host != "" {
+		port = portOnly(ctx.Request.Host)
+	} else {
+		port = portOnly(ctx.Request.URL.Host)
+	}
+
+	if isUsingNonDefaultPort(ctx.Request.URL.Scheme, port) {
+		ctx.Request.Host = hostname(ctx.Request) + ":" + port
+	} else if port != "" { // remove standard port, if set explicitly
+		ctx.Request.Host = hostname(ctx.Request)
+	}
+}
+
+// Returns host without port number
+func hostname(req *http.Request) string {
+	if req == nil {
+		return ""
+	}
+
+	if req.Host != "" {
+		return stripPort(req.Host)
+	}
+
+	return aws.URLHostname(req.URL)
+}
+
+// Copy of url.stripPort()
```

Comment:
could you update this to mention directly that this was copied for the Go standard library `url` package.  Would also be also to include the go version too.